Source/Core/Common/Config/ActiveInfos.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <unordered_set>
+#include <vector>
+
+#include "Common/Config/Config.h"
+#include "Common/Config/Layer.h"
+#include "Common/Config/Location.h"
+
+namespace Config
+{
+template <typename T>
+class Info;
+
+class ActiveInfosBase
+{
+public:
+  static void InvalidateCache();
+
+protected:
+  // unique_ptr is used to avoid problems with initialization order.
+  // We should be able to stop using unique_ptr once we are using C++20 everywhere,
+  // since C++20 makes std::vector's default constructor constexpr.
+  static std::unique_ptr<std::vector<void (*)()>> m_invalidate_cache_functions;
+
+  static bool m_program_is_exiting;
```

Comment:
Please don't prefix static variables with `m_`, that's very confusing. In fact, if you can, please avoid static variables at all...

What exactly does this `m_program_is_exiting` protect against, anyway?

Reply:
Ah, I wasn't sure whether to use `s_` or `m_` for static class members. I'll change it to `s_` then.

`ActiveInfos::Remove` can get called from the destructor of a `ConfigInfo` after the destructor of `m_active_infos` runs, so I added `m_program_is_exiting` so that `ActiveInfos::Remove` can check whether it's okay to access `m_active_infos`.